libcni/api.go

```diff
@@ -193,24 +209,77 @@ func (c *CNIConfig) getResultCacheFilePath(netName string, rt *RuntimeConf) stri
 	return filepath.Join(c.getCacheDir(rt), "results", fmt.Sprintf("%s-%s-%s", netName, rt.ContainerID, rt.IfName))
 }
 
-func (c *CNIConfig) setCachedResult(result types.Result, netName string, rt *RuntimeConf) error {
+func (c *CNIConfig) setCachedResult(result types.Result, config []byte, netName string, rt *RuntimeConf) error {
+
+	cached := cachedInfo{
+		Config: config,
+		Kind:   CNICacheV1}
+	if len(rt.Args) > 0 {
```

Comment:
This could provide distinction between an empty slice and non defined slice (`[]` vs `nil`). That could be usable if we would annotations for `RuntimeConf` for marshaling, but we don't have them, so also IMO this (and later `CapabilityArgs`) could simply go above struct.

Reply:
Agreed.  I don't want our API to depend on the difference between `[]` and `nil`.